twine/repository.py

```diff
@@ -153,6 +153,11 @@ def upload(self, package, max_redirects=5):
         return resp
 
     def package_is_uploaded(self, package, bypass_cache=False):
+        # NOTE(sigmavirus24): Not all indices are PyPI and pypi.io doesn't
+        # have a similar interface for finding the package versions.
+        if not self.url.startswith('https://pypi.python.org/pypi/'):
```

Comment:
If you upload to ``upload.pypi.io`` the change will be reflected on Legacy PyPI almost immediately (Warehouse purges Legacy). If you upload to ``pypi.python.org`` the change will be reflected on Warehouse *eventually*, but that take a large amount of time, a day or more (Legacy does not purge Warehouse).

Thus it's completely reasonable to upload to Warehouse by default, but you want to pull information from legacy until nobody is uploading to legacy any more. Further, at some point uploading to legacy will break, but pulling from legacy will always work because we can redirect a ``GET`` request to the Warehouse URL.